### PR TITLE
Use GitHub result file in regression test

### PR DESCRIPTION
**Pull request by rical**
Use the new fancy result-gh.md file which as github emojis.
Also,  split result publishing to its own run command to simplify the YAML file.

### CODE REVIEW

File: .github/workflows/regression.yml [hunk 1]
```diff
@@ -61,9 +61,9 @@ jobs:
...
       - name: Regression Test
-        run: |
-          rc=true
-          make test-qeneth || rc=false
-          cat test/.log/last/result.md >> $GITHUB_STEP_SUMMARY
-          eval $rc
+        run: make test-qeneth
+
+      - name: Publish Test Result
+        run: cat test/.log/last/result-gh.md >> $GITHUB_STEP_SUMMARY
```
Comment (troglobit) [inline .github/workflows/regression.yml:69]: Neat with more fancy log file! :rocket: 

However, all GitHub actions run with a sort of `set -e`, so with your refactor we'll never get the test results published if any test fails.

**Review by troglobit — CHANGES_REQUESTED** [review]
See review comment inline.

File: .github/workflows/regression.yml (hunk 1 above)
Comment (rical) [reply .github/workflows/regression.yml]: > Neat with more fancy log file! 🚀
> 
> However, all GitHub actions run with a sort of `set -e`, so with your refactor we'll never get the test results published if any test fails.

Great point. Didn't know that. I added the magical GitHub syntax `if: always()` which ensures the test results are published even if previous jobs fails.

Here's an example from the test framework:
https://github.com/rical/9pm/actions/runs/6377739586

**Review by troglobit — APPROVED** [review]

File: .github/workflows/regression.yml (hunk 1 above)
Comment (troglobit) [reply .github/workflows/regression.yml]: Nice!  Now I can fix a few of my other actions as well, thanks :smiley::+1:

**Comment by troglobit** [comment]
Merging despite regression test failure.  This problem is common to the main branch and tracked in #151.

**Pull request merged**